source/preprocessor.tex

```diff
@@ -743,20 +743,31 @@
 \pnum
 After the arguments for the invocation of a function-like macro have
 been identified, argument substitution takes place.
-A parameter in the replacement list, unless preceded by a
+A parameter in the replacement list --- unless it is
+\mname{VA_OPT}\tcode{(}\placeholder{content}\tcode{)}, preceded by a
 \tcode{\#}
 or
 \tcode{\#\#}
-preprocessing token or followed by a
+preprocessing token, or followed by a
 \tcode{\#\#}
-preprocessing token (see below),
+preprocessing token (see below) ---
 is replaced by the corresponding argument after all macros
-contained therein have been expanded.
-Before being substituted,
+contained therein have been expanded, in which case,
```

Comment:
I don't like this "in which case"; I think it makes the wording more confusing, because it could be referring to the case where the "unless" part holds or the case where it doesn't hold. For example, the more natural construction "X does Y unless Z, in which case Q" would mean that X does Y if !Z and Q happens if Z. I also don't like the condition for this being hyphened into the middle of the sentence. How about this instead:

> For each parameter in the replacement list that is neither preceded by a `#` or `##` preprocessing token nor followed by a `##` preprocessing token, the preprocessing tokens naming the parameter are replaced by a token sequence determined as follows:
>  * If the parameter is of the form `__VA_OPT__(`*content*`)`, the replacement preprocessing tokens are the preprocessing token sequence for the corresponding argument.
>  * Otherwise, the replacement preprocessing tokens are the preprocessing tokens of corresponding argument after all macros contained therein have been expanded. The argument's preprocessing tokens are completely macro replaced before being substituted as if they formed the rest of the preprocessing file with no other preprocessing tokens being available.
>
> [ *Example*: ... ]

Reply:
I like that. I've left the open sentence of the containing paragraph intact and then applied this change.

@hubert-reinterpretcast: any opinions?